raianalysis/raianalysis/raianalyzer/raianalyzer.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Defines the RAIAnalyzer class."""
+
+class RAIAnalyzer(object):
+    def __init__(self, model, train, test, target_column, transformation, task_type):
+        self.model = model
+        self.train = train
+        self.test = test
+        self.target_column = target_column
+        self.transformation = transformation
+        self.task_type = task_type
+
+    def add_fairness(self, sensitive_features):
+        pass
+
+    def add_explainer(self):
+        pass
+
+    def add_causal(self):
+        pass
+
+    def add_counterfactual(self,
+                           continuous_features=None,
+                           outcome_name=None,
+                           total_CFs=None, 
+                           desired_range=None,
+                           desired_class="opposite",
+                           permitted_range=None, 
+                           features_to_vary=None,
+                           method=None):
+        pass
+
+    def add_erroranalysis(self):
+        pass
+
+    def compute(self):
+        pass
+
+    def list(self):
+        pass
+
+    def save(self, path):
+        pass
+
+    @staticmethod
+    def load(path):
```

Comment:
This is a potential problem. If this is going to load a fully reusable `RAIAnalyzer` then the saved file is going to have to embed the `model`.

Reply:
maybe we can take in a model serializer object that defines how to serialize the model?  Otherwise we can allow the user to set the model on the RAIAnalyzer.

Reply:
will leave this question to a future PR, as I won't implement save/load in this PR since it's already quite large for the functionality it already has